performant-software/neatline-spa
# Exhibit create/read/destroy/update through API with JWT authentication

### What does this PR do?
- Retrieves a JWT set by the host view and uses it with API requests
- Restructures application around redux and router
- Supports exhibit CRUD operations through redux reducers
- Adds views to create and edit exhibits using redux-form
- Allows authenticated admin users to delete exhibits

### What issues does it address?
- Closes #5 
- Closes #7 

### How to test
- Go to https://neatline-3-staging.herokuapp.com/s/public-site/neatline to view the base Neatline page
- (For the same functionality without the Omeka site container, go to https://neatline-3-staging.herokuapp.com/s/public-site/neatline/full — this full window view is a work in progress in the backend app)
- Sign in as an admin at https://neatline-3-staging.herokuapp.com/admin, then return to the Neatline page
- Click "Create an Exhibit" and fill in values for the various fields, then submit the form with the button at the bottom of the page
- You should be returned to the Browse Exhibits page, and your newly created exhibit should appear at the bottom of the list
- Clicking on the exhibit's title should take you to a map viewer page with an exhibit settings sidebar. Note that the map itself doesn't yet respond to the exhibit settings, and will show an OpenStreetMaps base layer regardless of what you've set the layers to.
- Change some settings and click Save. If you change the exhibit slug, you should see the browser window URL automatically update to reflect the new slug when you save the exhibit.
- Return to the Exhibit Browse page by clicking the "Neatline" link at the top of the sidebar or using your browser's back button
- Click "Delete" for an exhibit. It should be removed from the list
